### PR TITLE
bpo-15954: Check return code of wcsxfrm().

### PR DESCRIPTION
**Pull request by serhiy-storchaka**

### CODE REVIEW

File: Modules/_localemodule.c [hunk 1]
```diff
@@ -269,7 +274,12 @@ PyLocale_strxfrm(PyObject* self, PyObject* args)
...
         buf = new_buf;
+        errno = 0;
         n2 = wcsxfrm(buf, s, n2+1);
+        if (errno) {
+            PyErr_SetFromErrno(PyExc_OSError);
+            goto exit;
+        }
     }
     result = PyUnicode_FromWideChar(buf, n2);
 exit:
```
Comment (vstinner) [inline Modules/_localemodule.c:285]: While you modify this function, can you please remove useless "if (buf)" and "if (s)" below? PyMem_Free(NULL) is well defined: it does nothing.

**Review by vstinner — APPROVED** [review]
LGTM except of a minor request.

File: Modules/_localemodule.c (hunk 1 above)
Comment (serhiy-storchaka) [reply Modules/_localemodule.c:285]: Except adding an overhead for function call.

But I don't think it matters in comparison with the time of calling other functions.

**Comment by vstinner** [comment]
> Except adding an overhead for function call.

I expect C calls to be *very cheap*. (In Python, function calls are
very expensive, it's the opposite ;-))

**Pull request merged**